### PR TITLE
BLD: try to build numpy instead of using the one on TravisCI

### PR DESCRIPTION
**Pull request by argriffing**
This is an experimental TravisCI related PR.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -44,7 +44,7 @@ matrix:
         - TESTMODE=fast
         - COVERAGE=
         - NPY_RELAXED_STRIDES_CHECKING=1
-        - NUMPYSPEC="git+git://github.com/numpy/numpy.git@v1.9.1"
+        - NUMPYSPEC="--upgrade git+git://github.com/numpy/numpy.git@v1.9.1"
```
Comment (ev-br) [inline .travis.yml]: This is going to quietly break again as soon as numpy 1.10 is out. Or maybe it is already broken, if preinstalled numpy is 1.9.2...

Comment (ev-br) [inline .travis.yml]: Maybe the right thing to do is to turn the diagnostic print below into an actual assertion?

Comment (argriffing) [inline .travis.yml]: @ev-br I see that you're right; I had the wrong idea about what the `--upgrade` flag does.

Comment (argriffing) [inline .travis.yml]: @ev-br Would you suggest `--force-reinstall` and/or `--ignore-installed` instead?  Regarding turning the diagnostic print into an actual assertion, do you mean something involving bash or do you mean adding a scipy unit test?  Also feel free to make a replacement PR; I'm just messing around with this one and I'd be OK to close it.

**Comment by argriffing** [comment]
This does what I'd hoped, but suggestions for making it more robust are welcome.

**Comment by ev-br** [comment]
Great! let's put this in right away then

File: .travis.yml (hunk 1 above)
Comment (ev-br) [inline .travis.yml]: TBH I've no idea --- the ways of pip are a mystery to me :-). 

All I'm guessing is that pinning a version to build is OK if augmented by either a nosetest `assert that strides are relaxed` or maybe there is a lighter touch solution to convert a python assertion into a bash level return code.

**Pull request merged**